### PR TITLE
Remove checksum and add crc32 support.

### PR DESCRIPTION
**Pull request by cheuschkel**

### CODE REVIEW

File: yasp.c [hunk 1]
```diff
@@ -45,16 +46,13 @@ int16_t process_yasp(uint8_t *buffer, uint16_t length)
...
     {
         return RET_CMD_INCOMPLETE;
     }
-    for (i = START_CHECKSUM; i < cmd_len + START_CHECKSUM; i++)
-    {
-        checksum += buffer[i];
-    }
-    //printf("Actual: %02x, Checksum: %02x\n", buffer[cmd_len + START_CHECKSUM], checksum);
-    if (checksum != buffer[cmd_len + START_CHECKSUM])
+    crc_32 = (crc_32, &buffer[START_CRC32], cmd_len);
```
Comment (mjbrown) [inline yasp.c]: What is going on here?  Shouldn't this be crc32(crc_32, &buffer[START_CRC32], cmd_len)?

File: yasp.c [hunk 2]
```diff
@@ -68,30 +66,27 @@ int16_t process_yasp(uint8_t *buffer, uint16_t length)
...
-    uint8_t checksum = 0x00;
+    uint32_t crc_32 = 0x00;
     out_buffer[SYNCH1_POS] = 0xFF;
     out_buffer[SYNCH2_POS] = 0xFF;
     out_buffer[LENGTH_POS] = (uint8_t)((length + 3) >> 8);
     out_buffer[LENGTH_POS+1] = (uint8_t)(length+3);
     out_buffer[COMMAND_POS] = 0x80 | cmd;
     serial_tx(out_buffer, 5);
-    checksum += out_buffer[2] + out_buffer[3] + out_buffer[4];
+    crc_32 = (crc_32, &out_buffer[2], 3);
```
Comment (mjbrown) [inline yasp.c]: Same here, the tests probably pass because you have the same mistake in two places.  Seems like we need a failing test case.

**Comment by cheuschkel** [comment]
Fixed crc32() calls.

**Comment by cheuschkel** [comment]
@mjbrown Can you check out the updated PR?

**Comment by cheuschkel** [comment]
Updated PR based on comments.  I am going crazy trying to figure out why `TestDoubleCommand()` does not work though..can you please review and let me know what you think?

**Pull request merged**